frontend/packages/console-app/src/components/user-preferences/UserPreferenceCheckboxField.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Checkbox, Skeleton } from '@patternfly/react-core';
+import { UserPreferenceCheckboxField as CheckboxFieldType } from '@console/dynamic-plugin-sdk/src';
+import { useUserSettings } from '@console/shared';
+import { UserPreferenceFieldProps } from './types';
+
+type UserPreferenceCheckboxFieldProps = UserPreferenceFieldProps<CheckboxFieldType>;
+
+const UserPreferenceCheckboxField: React.FC<UserPreferenceCheckboxFieldProps> = ({
+  id,
+  label,
+  userSettingsKey,
+  trueValue,
+  falseValue,
+}) => {
+  // types
+  type UserPreferenceValueType = typeof trueValue | typeof falseValue;
+
+  // resources and calls to hooks
+  const [
+    currentUserPreferenceValue,
+    setCurrentUserPreferenceValue,
+    currentUserPreferenceValueLoaded,
+  ] = useUserSettings<UserPreferenceValueType>(userSettingsKey);
+
+  const loaded: boolean = currentUserPreferenceValueLoaded;
+
+  // utils and callbacks
+  const getCheckboxValue = (checked: boolean): UserPreferenceValueType =>
+    checked ? trueValue : falseValue;
+
+  const onChange = (checked) => {
+    const checkedValue: UserPreferenceValueType = getCheckboxValue(checked);
+    checkedValue !== currentUserPreferenceValue && setCurrentUserPreferenceValue(checkedValue);
+  };
```

Comment:
Could be simplified into one function. Value is more needed for the onChange argument where the compiler could not know it then for the const value .... where the type is known based on the type of `? trueValue : falseValue`.

```suggestion
  const onChange = (checked: boolean) => {
    const value = checked ? trueValue : falseValue;
    if (value !== currentUserPreferenceValue) {
      setCurrentUserPreferenceValue(value);
    }
  };
```